src/Functions/DateTimeTransforms.h

```diff
@@ -1446,26 +1479,34 @@ struct Transformer
     }
 };
 
-
 template <typename FromDataType, typename ToDataType, typename Transform, bool is_extended_result = false>
 struct DateTimeTransformImpl
 {
+    template <typename Additions = void *>
     static ColumnPtr execute(
         const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t /*input_rows_count*/, const Transform & transform = {})
     {
-        using Op = Transformer<typename FromDataType::FieldType, typename ToDataType::FieldType, Transform, is_extended_result>;
+        using Op = Transformer<FromDataType, ToDataType, Transform, is_extended_result, Additions>;
 
         const ColumnPtr source_col = arguments[0].column;
         if (const auto * sources = checkAndGetColumn<typename FromDataType::ColumnType>(source_col.get()))
         {
+            ColumnUInt8::MutablePtr col_null_map_to;
```

Comment:
Help me understand: the code added here creates a bitmap for result rows that are null. So that affects only functions `to[...]OrNull()`, right? Did they not work before?

Reply:
- `toDate` works through `FunctionConvert / DateTimeTransformImpl (specialization of ConvertImpl) / `
- `toDateOrDefault` works through `FunctionCastOrDefault / castColumnAccurateOrNull / FunctionCast (i add new code here) `
- `toDateOrNull` works through `FunctionConvertFromString`  and not fixed (not accepted Int as argument)
- `accurateCast / accurateCastOrNull`() works through `FunctionCast` and fixed.

DateTimeTransformImpl (before) supported only implicit  cast form (= not accurate = CAST())